SIGBUS: Illegal storage access. (Attempt to read from nil?) on macOS

```
❯ curl https://nim-lang.org/choosenim/init.sh -sSf | sh
choosenim-init: Downloading choosenim-0.4.0_macosx_amd64
SIGBUS: Illegal storage access. (Attempt to read from nil?)
```
on zsh/bash on macOS 10.14.2

Hmm - it could be - here is my version:
```
❯ openssl version
OpenSSL 1.1.0j  20 Nov 2018
```

At the end of httpclient.nim, I added:
```
when isMainModule:
  let j = getDefaultSSL()
```
with no error when I run it.

When I check the  the wrapper (`lib/wrappers/openssl.nim`):

```nim
  when defined(osx):
    const versions = "(.1.1|.38|.39|.41|.43|.44|.45|.46|.10|.1.0.2|.1.0.1|.1.0.0|.0.9.9|.0.9.8|)"
  else:
    const versions = "(.1.1|.1.0.2|.1.0.1|.1.0.0|.0.9.9|.0.9.8|.46|.45|.44|.43|.41|.39|.38|.10|)"
```

When I build and test the wrapper:
```
❯ ./bin/nim c -d:ssl -p:. -r tests/untestable/tssl.nim
Hint: used config file '/Users/aljohnson/code/Nim/config/nim.cfg' [Conf]
Hint: used config file '/Users/aljohnson/code/Nim/config/config.nims' [Conf]
Hint: system [Processing]
Hint: widestrs [Processing]
Hint: io [Processing]
Hint: tssl [Processing]
Hint: httpclient [Processing]
Hint: net [Processing]
Hint: nativesockets [Processing]
Hint: os [Processing]
Hint: strutils [Processing]
Hint: parseutils [Processing]
Hint: math [Processing]
Hint: bitops [Processing]
Hint: macros [Processing]
Hint: algorithm [Processing]
Hint: unicode [Processing]
Hint: pathnorm [Processing]
Hint: osseps [Processing]
Hint: posix [Processing]
Hint: times [Processing]
Hint: options [Processing]
Hint: typetraits [Processing]
Hint: sets [Processing]
Hint: hashes [Processing]
Hint: openssl [Processing]
Hint: dynlib [Processing]
Hint: uri [Processing]
Hint: strtabs [Processing]
Hint: base64 [Processing]
Hint: mimetypes [Processing]
Hint: random [Processing]
Hint: httpcore [Processing]
Hint: tables [Processing]
Hint: streams [Processing]
Hint: asyncnet [Processing]
Hint: asyncdispatch [Processing]
Hint: heapqueue [Processing]
Hint: lists [Processing]
Hint: asyncstreams [Processing]
Hint: asyncfutures [Processing]
Hint: deques [Processing]
Hint: cstrutils [Processing]
Hint: selectors [Processing]
Hint: kqueue [Processing]
Hint: asyncfile [Processing]
CC: stdlib_assertions.nim
CC: stdlib_io.nim
CC: stdlib_system.nim
CC: stdlib_parseutils.nim
CC: stdlib_math.nim
CC: stdlib_strutils.nim
CC: stdlib_pathnorm.nim
CC: stdlib_posix.nim
CC: stdlib_options.nim
CC: stdlib_times.nim
CC: stdlib_os.nim
CC: stdlib_nativesockets.nim
CC: stdlib_hashes.nim
CC: stdlib_sets.nim
CC: stdlib_dynlib.nim
CC: stdlib_openssl.nim
CC: stdlib_net.nim
CC: stdlib_uri.nim
CC: stdlib_base64.nim
CC: stdlib_random.nim
CC: stdlib_tables.nim
CC: stdlib_httpcore.nim
CC: stdlib_streams.nim
CC: stdlib_heapqueue.nim
CC: stdlib_deques.nim
CC: stdlib_asyncfutures.nim
CC: stdlib_kqueue.nim
CC: stdlib_selectors.nim
CC: stdlib_asyncdispatch.nim
CC: stdlib_httpclient.nim
CC: tssl.nim
Hint:  [Link]
Hint: operation successful (77530 lines compiled; 2.348 sec total; 90.57MiB peakmem; Debug Build) [SuccessX]
Hint: /Users/aljohnson/code/Nim/tests/untestable/tssl  [Exec]
version: 0x00000000101000AF
done
```

Test:

```
❯ ./bin/nim c -d:ssl -p:. --dynlibOverride:ssl --passL:-lcrypto --passL:-lssl -r tests/untestable/tssl.nim
Hint: used config file '/Users/aljohnson/code/Nim/config/nim.cfg' [Conf]
Hint: used config file '/Users/aljohnson/code/Nim/config/config.nims' [Conf]
Hint: system [Processing]
Hint: widestrs [Processing]
Hint: io [Processing]
Hint: tssl [Processing]
Hint: httpclient [Processing]
Hint: net [Processing]
Hint: nativesockets [Processing]
Hint: os [Processing]
Hint: strutils [Processing]
Hint: parseutils [Processing]
Hint: math [Processing]
Hint: bitops [Processing]
Hint: macros [Processing]
Hint: algorithm [Processing]
Hint: unicode [Processing]
Hint: pathnorm [Processing]
Hint: osseps [Processing]
Hint: posix [Processing]
Hint: times [Processing]
Hint: options [Processing]
Hint: typetraits [Processing]
Hint: sets [Processing]
Hint: hashes [Processing]
Hint: openssl [Processing]
Hint: dynlib [Processing]
Hint: uri [Processing]
Hint: strtabs [Processing]
Hint: base64 [Processing]
Hint: mimetypes [Processing]
Hint: random [Processing]
Hint: httpcore [Processing]
Hint: tables [Processing]
Hint: streams [Processing]
Hint: asyncnet [Processing]
Hint: asyncdispatch [Processing]
Hint: heapqueue [Processing]
Hint: lists [Processing]
Hint: asyncstreams [Processing]
Hint: asyncfutures [Processing]
Hint: deques [Processing]
Hint: cstrutils [Processing]
Hint: selectors [Processing]
Hint: kqueue [Processing]
Hint: asyncfile [Processing]
CC: stdlib_system.nim
CC: stdlib_strutils.nim
CC: stdlib_openssl.nim
CC: stdlib_net.nim
Hint:  [Link]
Undefined symbols for architecture x86_64:
  "_OPENSSL_init_ssl", referenced from:
      _SSL_library_init_6SfBfS64wp6JZihQGGrnKg in stdlib_openssl.nim.c.o
  "_OpenSSL_version_num", referenced from:
      _getOpenSSLVersion_dJsO04BYfgh1GDCx4NNEgA in stdlib_openssl.nim.c.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: execution of an external program failed: 'clang   -o /Users/aljohnson/code/Nim/tests/untestable/tssl  /Users/aljohnson/.cache/nim/tssl_d/stdlib_assertions.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_io.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_system.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_parseutils.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_math.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_strutils.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_pathnorm.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_posix.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_options.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_times.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_os.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_nativesockets.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_hashes.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_sets.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_openssl.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_net.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_uri.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_base64.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_random.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_tables.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_httpcore.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_streams.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_heapqueue.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_deques.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_asyncfutures.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_kqueue.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_selectors.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_asyncdispatch.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/stdlib_httpclient.nim.c.o /Users/aljohnson/.cache/nim/tssl_d/tssl.nim.c.o  -lm  -lcrypto -lssl  -ldl'
```

which looks like it could be part of the issue. I'll try changing some openssl stuff and see whats going on...

--- a/src/choosenim.nims
+++ b/src/choosenim.nims
@@ -1,4 +1,4 @@
 when defined(macosx):
   --define:curl
-
---define:ssl
+else:
+  --define:ssl

